python/pyspark/ml/linalg/__init__.py

```diff
@@ -511,6 +519,12 @@ def __init__(self, size, *args):
                         "Indices %s and %s are not strictly increasing"
                         % (self.indices[i], self.indices[i + 1]))
 
+        assert np.max(self.indices) < self.size, \
```

Comment:
Will this disallow a size 0 vector? not sure what np.max does here.
PS I like its error message better for "not strictly increasing" and we should probably make the messages generally consistent.